presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorMetadata.java

```diff
@@ -266,6 +263,25 @@ default void finishDelete(ConnectorSession session, ConnectorTableHandle tableHa
     }
 
     /**
+     * Notifies when select query starts
+     *
+     * @return serializable {@link ConnectorTableHandle}. This handle will be brought back in {@link #finishSelect(ConnectorSession, ConnectorTableHandle)} if null returned, no finishSelect will be called.
+     */
+    default ConnectorTableHandle beginSelect(ConnectorSession session, ConnectorTableHandle tableHandle, Optional<ConnectorTableLayoutHandle> layoutHandle, Collection<ColumnHandle> columnHandles)
```

Comment:
But anyway, if we actually wanted to support this, we should have `beginSelect` return some sort of identifier that is passed back to `finishSelect` rather than swapping out the table handle.